### ISSUE TITLE
2.12 support: ApplicationSet Role  cannot list resource "appprojects" in API group  "argoproj.io"

### ISSUE THREAD

**Issue by eripa** [argoproj/argo-helm#2813]
### Describe the bug

After deploying the Argo CD 2.12.0-rc3, the following can be observed in the aplicationset controller:

```
E0708 16:19:38.760635       7 reflector.go:147] pkg/mod/k8s.io/client-go@v0.29.6/tools/cache/reflector.go:229: Failed to watch *v1a
lpha1.AppProject: failed to list *v1alpha1.AppProject: appprojects.argoproj.io is forbidden: User "system:serviceaccount:argo:argocd-applicationset-controller" cannot list resource "appprojects" in API group
 "argoproj.io" in the namespace "argo"
```

### Related helm chart

argo-cd

### Helm chart version

7.3.4

### To Reproduce

1. deploy Argo CD 2.12.0-rc3
2. observe behaviour in applicationset-controller

### Expected behavior

The ApplicationSet controller should have the necessary permissions to operate normally.

**Comment by yu-croco** [comment]
Hi @eripa , thank you for opening the issue.

> After deploying the Argo CD 2.12.0-rc3

We follow upstream's manifest and we don't follow rc versions.
So please wait until upstream releases new version.

When upstream releases, Renovate make PR in argo-helm and we handle manifest's update in same PR.

**Comment by yu-croco** [comment]
Ah @pdrastil already explained in the PR. Thank you!

**Comment by yu-croco** [comment]
https://github.com/argoproj/argo-helm/pull/2820 will resolve this issue. FYI

**Comment by hodorov** [comment]
@yu-croco #2820 don't resolve similar issue when appset controller deployed with **allowAnyNamespace** option
cluster role ([ref](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-applicationset/clusterrole.yaml#L34-L37)) sill have only "get" verbose

**Comment by hodorov** [comment]
Fixed in #2868

**Comment by yu-croco** [comment]
fixed in https://github.com/argoproj/argo-helm/pull/2885
